packages/cli/src/ActiveWebhooks.ts

```diff
@@ -102,9 +102,16 @@ export class ActiveWebhooks implements IWebhookManager {
 			settings: workflowData.settings,
 		});
 
-		const additionalData = await WorkflowExecuteAdditionalData.getBase(
-			workflowData.shared[0].user.id,
-		);
+		const ownerRelation = workflowData.shared
+			.filter((sw) => sw.role === 'workflow:owner')
+			.flatMap((sw) => sw.project.projectRelations)
+			.find((pr) => pr.role === 'project:personalOwner');
+
+		if (!ownerRelation) {
+			throw new ApplicationError(`Workflow ${workflowData.display()} has no owner`);
+		}
```

Comment:
Oh, nice. What prevents us from removing the `userId` completely from `getBase`? Is this used by the telemetry?

Reply:
Oh, that's not merged yet. Which means I 

1. either wait until https://github.com/n8n-io/n8n/pull/8589 is merged (maybe tomorrow already) and then update this PR,
2. or merge this PR first and then https://github.com/n8n-io/n8n/pull/8589 needs to be updated.

I think 1. may be better, because I have time to resolve conflicts tomorrow and update this PR for re-review.

WDYT?

Reply:
Yeah `userId` is still used in telemetry and. we'd have to keep it, hence it stays. But to be honest, we could perhaps extract this to a separate function that fetches the user ID based on the workflow ID, if it exists. Workflows in team project would return `undefined` as user ID. Something to discuss.

Reply:
I've just merged #8589 so you can merge and update your branch.

Reply:
This is the only place where it still exists: 
https://github.com/n8n-io/n8n/pull/8886/files?show-viewed-files=true&file-filters%5B%5D=#diff-cb54fea004c361fc209313b1a284e031c3cd9251634b08c663dbaaee51cfdae7R113-R123

I didn't want to replace it there with a call to the ownership service because this used to work sync and probably can stay sync code.

Apart from that it was mostly used to get the user for `getBase`, but on feature/rbac all places where it used to pass a user it does not pass it anymore:

![image](https://github.com/n8n-io/n8n/assets/927609/8eb96f94-99f1-4d15-b3c2-edca850c1ef1)
![image](https://github.com/n8n-io/n8n/assets/927609/4c9af52a-b099-4d3a-9a6b-54d0505f3963)
![image](https://github.com/n8n-io/n8n/assets/927609/4bb1ab2c-a208-4a33-9807-9799202201af)
![image](https://github.com/n8n-io/n8n/assets/927609/2194a976-1ed3-4654-940e-c4ca8c553642)

See: https://github.com/n8n-io/n8n/compare/master...feature/rbac

Reply:
Which means I can probably remove 
https://github.com/n8n-io/n8n/blob/cbc2a2e61d1fe277a84542f66753f2131ee10f8f/packages/cli/src/services/ownership.service.ts#L24-L57

Reply:
If there are no more calls to `getWorkflowOwnerCached` let's by all means remove it, we might have to add a different flavor of it but we can do this separately for Telemetry alone instead of having it being called in multiple places throughout the code. Perhaps we could change the `InternalHooks` file alone and fix this in a single place.

Reply:
Removed.